apps/cowswap-frontend/src/common/hooks/useAnalyticsReporter.ts

```diff
@@ -93,6 +96,15 @@ export function useAnalyticsReporter() {
     googleAnalytics.setDimension(Dimensions.market, marketDimension)
   }, [marketDimension])
 
+  useEffect(() => {
+    // Custom dimension 6 - injected widget app id
+    const injectedWidgetAppId = injectedWidgetMetaData
+      ? `${injectedWidgetMetaData.appKey}:${injectedWidgetMetaData.url}`
+      : ''
+
+    googleAnalytics.setDimension(Dimensions.injectedWidgetAppId, injectedWidgetAppId)
+  }, [injectedWidgetMetaData])
```

Comment:
This object won't change often, right? if that's the case, all good, but otherwise, i would suggest to make the hook depend on the URL and appKey directly

Reply:
Nice suggestion! Refactored